posthog/api/file_system.py

```diff
@@ -138,12 +139,18 @@ def safely_get_queryset(self, queryset: QuerySet) -> QuerySet:
         if self.action == "list":
             queryset = queryset.order_by("path")
 
+        if self.user_access_control:
+            queryset = self.user_access_control.filter_and_annotate_file_system_queryset(queryset)
```

Comment:
These changes in the viewset look good but I do wonder if the access control filter should be moved into the model manager so you don't have to think about it. Something to ponder. I just worry if it needs to be used in every location then it could get forgotten.

Reply:
I do wonder the same. I just added a `count_by_path` endpoint, and had to now merge with master and add the access control check manually. 

I did think about making this check automatic, but decided against it because I wasn't 100% sure of the Django API here and of all the places to add it. I was afraid of accidentally missing something, and causing leakage tha way. I guess this is good enough for now?

Reply:
Yeah good for now but worth relooking at.

Bascially just add a custom manager to the FileSystem model that overwrites the queryset. The tricky part would be making sure the manager has access to the user access control which it might not be possible. 

Quick psuedo code:
```
class FileSystemManager(RootTeamManager):
    def get_queryset(self):
        qs = super().get_queryset()
        qs = self.user_access_control.filter_and_annotate_file_system_queryset(qs)
        return qs

class FileSystem(models.Model):
    objects = FileSystemManager()
    
    ...
    
```